File: Scripts/scrape_python/coolmath.py
import requests
from bs4 import BeautifulSoup as bs
import json
import concurrent.futures

from utils import *

def coolmath():
    output = []

    url = "https://www.coolmathgames.com/1-complete-game-list/view-all"
    page = requests.get(url)

    if debug:
        print("Request succeded")

    soup = bs(page.content, "html.parser")
    # find all game links
    games = soup.find_all(
        "div", class_="view-content")[2].find_all("span", class_="game-title")

    for game in games:
        link = game.contents[0]
        gameurl = "https://www.coolmathgames.com" + link["href"] + "/play"

        if game.parent.contents[-1]["class"][0] == 'icon-gamethumbnail-all-game-pg':
            continue

        if not exists(gameurl):
            continue

        name = link.get_text()

        output.append([name, gameurl])
        if debug:
            print([name, gameurl])
    return output

File: Scripts/scrape/sites/coolmath.js
import { smart_get, exists, processResult } from "../utils.js";
import { load } from "cheerio";

export const coolmath = async () => {
	const results = [];

	const promises = [];
	const res = await smart_get(
		"https://www.coolmathgames.com/1-complete-game-list/view-all"
	);

	console.log("coolmath request done");
	const $ = load(res.data);

	$(".view-all-games:first")
		.find("> .views-row:not(:has(>.icon-gamethumbnail-all-game-pg))")
		.find(".game-title a:first-child")
		.each((_, elem) => {
			promises.push(
				(async (elem) => {
					const gametext = $(elem).text();
					const gamehref = $(elem).attr("href");
					const gameurl = `https://www.coolmathgames.com${gamehref}/play`;
					const pageurl = `https://www.coolmathgames.com${gamehref}`;

					if (await exists(gameurl)) {
						processResult(results, [gametext, gameurl], "Coolmath");
					} else if (await exists(pageurl)) {
						processResult(
							results,
							[gametext, pageurl],
							"Coolmath page"
						);
					}
				})(elem)
			);
		});
	await Promise.all(promises);
	console.log("done with coolmath");
	return results;
};
